include/deal.II/fe/fe_values.h

```diff
@@ -2065,6 +2065,94 @@ class FEValuesBase :
   const std::vector<DerivativeForm<2,dim,spacedim> > &get_jacobian_grads () const;
 
   /**
+   * Return the second derivative of the transformation from unit to real
+   * cell, i.e. the first derivative of the Jacobian, at the specified
+   * quadrature point, pushed forward to the real cell coordinates, i.e.
+   * $G_{ijk}=dJ_{iJ}/d\hat x_K (J_{jJ})^{-1} (J_{kK})^{-1}$.
+   *
+   * @dealiiRequiresUpdateFlags{update_jacobian_pushed_forward_grads}
+   */
+  const Tensor<3,spacedim> &jacobian_pushed_forward_grad (const unsigned int quadrature_point) const;
+
+  /**
+   * Return a reference to the array holding the values returned by
+   * jacobian_pushed_forward_grads().
+   *
+   * @dealiiRequiresUpdateFlags{update_jacobian_pushed_forward_grads}
+   */
+  const std::vector<Tensor<3,spacedim> > &get_jacobian_pushed_forward_grads () const;
+
+  /**
+   * Return the third derivative of the transformation from unit to real
+   * cell, i.e. the second derivative of the Jacobian, at the specified
+   * quadrature point, i.e. $G_{ijkl}=\frac{d^2J_{ij}}{d\hat x_k d\hat x_l}$.
+   *
+   * @dealiiRequiresUpdateFlags{update_jacobian_hessians}
+   */
+  const DerivativeForm<3,dim,spacedim> &jacobian_hessian (const unsigned int quadrature_point) const;
```

Comment:
Thanks, I like the idea about using mapping_nth_derivative rather than hessians and gradients. I'll implement the name changes.